features/admin.actions.v1/models/actions.ts

```diff
@@ -0,0 +1,343 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Different Authentication Types.
+ */
+export enum AuthenticationType {
+    NONE = "NONE",
+    BASIC = "BASIC",
+    API_KEY = "API_KEY",
+    BEARER = "BEARER",
+}
+
+/**
+ * Different Action Types.
+ */
+export enum ActionType {
+    PRE_ISSUE_ACCESS_TOKEN = "PRE_ISSUE_ACCESS_TOKEN",
+    PRE_UPDATE_PASSWORD = "PRE_UPDATE_PASSWORD",
+    PRE_UPDATE_PROFILE = "PRE_UPDATE_PROFILE",
+    PRE_REGISTRATION = "PRE_REGISTRATION"
+}
+
+/**
+ * Action Status.
+ */
+enum ActionStatus {
+    ACTIVE,
+    INACTIVE
+}
+
+/**
+ * Interface for the authentication type dropdown options.
+ */
+export interface AuthTypeDropdownOption {
+    key: AuthenticationType;
+    text: string;
+    value: AuthenticationType;
+}
+
+/**
+ *  Action configuration.
+ */
+export interface ActionInterface {
+    /**
+     * Name of the Action.
+     */
+    name: string;
+    /**
+     * Description of the Action.
+     */
+    description?: string;
+    /**
+     * Endpoint configuration of the Action.
+     */
+    endpoint: EndpointInterface;
+}
+
+/**
+ *  Endpoint configuration.
+ */
+export interface EndpointInterface {
+    /**
+     * External endpoint.
+     */
+    uri: string;
+    /**
+     * Authentication configurations of the Action.
+     */
+    authentication: AuthenticationInterface;
+}
+
+/**
+ *  Endpoint authentication configuration.
+ */
+export interface AuthenticationInterface {
+    /**
+     * Authentication Type.
+     */
+    type: AuthenticationType;
+    /**
+     * Authentication properties.
+     */
+    properties: AuthPropertiesInterface;
+}
+
+/**
+ * Authentication Properties.
+ */
+export interface AuthPropertiesInterface {
+    /**
+     * Username auth property.
+     */
+    username?: string;
+    /**
+     * Password auth property.
+     */
+    password?: string;
+    /**
+     * Access Token auth property.
+     */
+    accessToken?: string;
+    /**
+     * Header auth property.
+     */
+    header?: string;
+    /**
+     * Value auth property.
+     */
+    value?: string;
+}
+
+/**
+ *  API Key Authentication Properties.
+ */
+export interface ApiKeyAuthPropertiesInterface {
+    /**
+     * Header auth property.
+     */
+    header: string;
+    /**
+     * Value auth property.
+     */
+    value: string;
+}
+
+/**
+ *  Action Basic Response.
+ */
+export interface ActionBasicResponseInterface {
+    /**
+     * ID of the Action.
+     */
+    id?: string;
+    /**
+     * Type of the Action.
+     */
+    type?: ActionType;
+    /**
+     * Name of the Action.
+     */
+    name?: string;
+    /**
+     * Description of the Action.
+     */
+    description?: string;
+    /**
+     * Status of the Action.
+     */
+    status?: ActionStatus;
+}
+
+/**
+ *  Action Response.
+ */
+export interface ActionResponseInterface {
+    /**
+     * ID of the Action.
+     */
+    id?: string;
```

Comment:
use and operator and user above interface

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/6722/commits/d96314b591f6f6f33b709b0600c399b789e7fcef